### PR TITLE
fix: revert renaming of query param 'match' -> 'partialMatch'

### PR DESCRIPTION
**Pull request by aakropotkin**
## Overview
Restore `pkgdb search` parameter _name_ `query.match`.

## Context
When adding separate internal arguments for `partialMatch` and `pnameOrPkgAttrName` from the existing `match` argument, the _public interface_ was also renamed from `match` to `partialMatch`.

The _public interface_ change was unintended, and caused breakage in existing usage.

### CODE REVIEW

**Review by zmitchell — APPROVED** [review]

File: src/search/params.cc [hunk 1]
```diff
@@ -30,16 +30,67 @@ SearchQuery::clear()
...
+          getOrFail( key, value, qry.semver );
+        }
+      else if ( key == "match" )
+        {
+          getOrFail( key, value, qry.partialMatch );
+        }
+      else
+        {
+          throw FloxException(
+            "Unrecognized search query key: 'query." + key + "'."
```
Comment (mkenigs) [inline src/search/params.cc:82]: Is there no easy way to do this with `nlohmann`? Just looked at the macros briefly, and it looks like they can do the equivalent of the `getOrFail` stuff, but maybe not throwing for an unrecognized field?

File: src/search/params.cc [hunk 2]
```diff
@@ -30,16 +30,67 @@ SearchQuery::clear()
   void
 from_json( const nlohmann::json & jfrom, SearchQuery & qry )
 {
-  pkgdb::from_json( jfrom, dynamic_cast<pkgdb::PkgDescriptorBase &>( qry ) );
-  try { jfrom.at( "partialMatch" ).get_to( qry.partialMatch ); }
-  catch( const nlohmann::json::out_of_range & ) {}
+  auto getOrFail = [&]( const std::string    & key
```
Comment (mkenigs) [inline src/search/params.cc:33]: (I don't feel like you need to move it, but I do think this change probably belongs in a separate PR)

**Review by mkenigs — COMMENTED** [review]
It seems like we're early enough in development that it might be more clear to just keep things consistent, especially since this is an interface for ourselves (i.e. the people writing `match` in Rust will be the same people having to write `partialMatch` in C++). Should we just keep it `match` everywhere? Or if `partialMatch` is more clear, I don't think that would be hard to change on the Rust side

File: src/search/params.cc (hunk 1 above)
Comment (aakropotkin) [reply src/search/params.cc:82]: Yeah they can if you use the `NLOHMANN_DEFINE_TYPE_*` macros, but those unfortunately will not get the "Unrecognized field" error.

With that in mind I needed to do a `from_json` _by hand_.

The `getOrFail` might not be necessary, but I wanted the messages to read _sort of like an attribute path_ so "failed to parse `query.name`: <nlohmann error messag>" instead of just the plain nlohmann error since it lacks context.

**Comment by mkenigs** [comment]
I also don't mind merging if it's blocking @zmitchell but would be curious to understand why we aren't just trying to keep things consistent

File: src/search/params.cc (hunk 2 above)
Comment (aakropotkin) [reply src/search/params.cc:33]: I added it here since I was replacing the `from_json` invocation on the base class and wanted to get nicer messages.

For similar cases I think we'll tackle those in the "better error handling" PR.
